extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcUtils.java

```diff
@@ -242,13 +243,12 @@ private static List<String> convertJsonArrayToList(JsonArray claimValue) {
     static QuarkusSecurityIdentity validateAndCreateIdentity(
             RoutingContext vertxContext, TokenCredential credential,
             TenantConfigContext resolvedContext, JsonObject tokenJson, JsonObject rolesJson, UserInfo userInfo,
-            TokenIntrospection introspectionResult) {
+            TokenIntrospection introspectionResult, Map<String, Object> requestData) {
 
         OidcTenantConfig config = resolvedContext.oidcConfig;
         QuarkusSecurityIdentity.Builder builder = QuarkusSecurityIdentity.builder();
         builder.addCredential(credential);
-        AuthorizationCodeTokens codeTokens = vertxContext != null ? vertxContext.get(AuthorizationCodeTokens.class.getName())
-                : null;
+        AuthorizationCodeTokens codeTokens = (AuthorizationCodeTokens) requestData.get(AuthorizationCodeTokens.class.getName());
```

Comment:
Hmm, but `CodeAuthenticationMechanism` passes them via `vertxContext`

Reply:
which is fine because it is same map and it is already in it because `CodeAuthenticationMechanism` runs before this method is invoked?

Reply:
All data are available at request and request data now. Please comment more if you see an issue. Thanks.

Reply:
Sure, lets continue tomorrow, I'm being forced to switch off my laptop :-)